nodes/all_in_one_publisher.py

```diff
@@ -124,17 +113,32 @@ def create_pipe(pipe_name):
                 sim_time_msg = json_message_converter.convert_json_to_ros_message('rosgraph_msgs/Clock', msg_list[1])
                 object_class.pub_sim.publish(sim_time_msg)
 
-            elif msg_list[0] == "nav_msgs/Odometry":
-                odom_msg = json_message_converter.convert_json_to_ros_message('nav_msgs/Odometry', msg_list[1])
-                object_class.pub_odom.publish(odom_msg)
-
-            elif msg_list[0] == "sensor_msgs/LaserScan":
-                scan_msg = json_message_converter.convert_json_to_ros_message('sensor_msgs/LaserScan', msg_list[1])
-                object_class.laser_scan_pub(scan_msg)
+            elif msg_list[1] == "nav_msgs/Odometry":
+                veh_ns = msg_list[0]
+                if veh_ns not in object_class.pub_odom_dic:
+                    topic_name = veh_ns + "/odom"
+                    create_publisher(veh_ns, object_class.pub_odom_dic, topic_name, Odometry)
```

Comment:
I'm not sure I understand these lines.

Where does `veh_ns` comes from? That's written to the pipe from the Lua side, isn't it? But isn't that the full topic name now?

If so, why do we still do `topic_name = veh_ns + "/odom"`?

Reply:
I haven't pushed the changes.

Reply:
Pushed!